src/layers/src/geojson-layer/geojson-layer.ts

```diff
@@ -429,11 +504,15 @@ export default class GeoJsonLayer extends Layer {
     const pickable = interactionConfig.tooltip.enabled;
     const hoveredObject = this.hasHoveredObject(objectHovered);
 
-    return [
-      new DeckGLGeoJsonLayer({
+    const {data, ...props} = dataProps;
+    const deckLayerData = this.isArrow ? data : [data];
+    const deckLayers = deckLayerData.map((d, i) => {
+      return new DeckGLGeoJsonLayer({
         ...defaultLayerProps,
         ...layerProps,
-        ...data,
+        ...props,
+        data: d,
+        id: this.isArrow ? `${this.id}-${i}` : this.id,
```

Comment:
Why an index? Is this related to batching? Maybe that should be a separate PR?

Reply:
Yes, this is related to batching in arrow table. Deck.gl will return this id for hovering.